src/highlight/HighlightAnnotations.tsx

```diff
@@ -97,6 +106,13 @@ const HighlightAnnotations = (props: Props): JSX.Element => {
                     />
                 </div>
             )}
+
+            {/* Layer 4: Annotations promoter to promote selection to staged */}
+            {!isCreating && selection && (
+                <div className="ba-HighlightAnnotations-promoterPopup">
+                    <PopupHighlight onClick={handlePromote} rect={selection.boundingRect} />
```

Comment:
It seems like the `boundingRect` can be derived from the `selection.rects` if we wanted to avoid storing that extra state